blog: ecosystem node 22 update blog (#698)
electron/website: blog/node-22-ecosystem-update.md

---
title: Moving our Ecosystem to Node 22
date: 2024-12-02T00:00:00.000Z
authors: VerteDinde
slug: ecosystem-node-22
tags: [community, ecosystem]
---

In early 2025, Electron’s npm ecosystem repos (under the `@electron/` and `@electron-forge/` namespaces) will move to Node.js 22 as the minimum supported version.

---

### What does this mean?

In the past, packages in Electron’s npm ecosystem (Forge, Packager, etc) have supported Node versions for as long as possible, even after a version has reached its End-Of-Life (EOL) date. This is done to make sure we don’t fragment the ecosystem—we understand that many projects depend on older versions of Node, and we don’t want to risk stranding those projects unless there was a pressing reason to upgrade.

> Over time, maintenance of this policy has become increasingly difficult for a few reasons:

Reviewer: This phrasing "maintenance of this policy" is a little confusing about whether the _version_ is changing (a one-time bump) or whether the _policy_ is changing too (such as, more frequent bumps in the future)

If it's the latter, then we should be a lot more explicit about it here. If it's the former, I'd rephrase:

```suggestion
Over time, using Node.js 14 as our minimum version has become increasingly difficult for a few reasons:
```

- Lack of official Node.js 14 macOS ARM64 builds requires us to maintain CI infrastructure workarounds to provide full test coverage.
- `engines` requirements for upstream package dependencies have moved forward, making it increasingly difficult to resolve supply chain security issues with dependency bumps.

> Additionally, newer versions of Node.js have included many improvements that we would like to leverage, such as runtime-native common utilities (e.g. [`fs.glob`](https://nodejs.org/api/fs.html#fsglobpattern-options-callback) and [`util.parseArgs`](https://nodejs.org/api/util.html#utilparseargsconfig)) and entire new batteries-included modules (e.g. [`node:test`](https://nodejs.org/api/test.html), [`node:sqlite`](https://nodejs.org/api/sqlite.html)).

Reviewer: I'm not sure about this one, but should this paragraph be a third bullet point in the previous list? Fits the rule of three

### Why upgrade now?

In July 2024, Electron’s Ecosystem Working Group decided to upgrade all packages to the earliest Node version where `require()`of synchronous ESM graphs will be supported (see [nodejs/node#51977](https://github.com/nodejs/node/pull/51977) and [nodejs/node#53500](https://github.com/nodejs/node/pull/53500)) at a future point after that version reaches its LTS date.

We’ve decided to set that update time to January/February 2025. After this upgrade occurs, Node 22 will be the minimum supported version in existing ecosystem packages.

### What action do I need to take?

We’ll strive to maintain compatibility as much as possible. However, to ensure the best support, we encourage you to upgrade your apps to Node 22 or higher.

> Note that the Node version running in your project is unrelated to the Node version embedded into your current version of Electron.

Reviewer: This is a good start but I'd go even further, e.g.  explaining if your app uses  Node 21 but doesn't use any ecosystem apps, they don't need to make any changes (I think?)

---
title: Moving our Ecosystem to Node 22
date: 2024-12-02T00:00:00.000Z
authors: VerteDinde
slug: ecosystem-node-22
tags: [community, ecosystem]
---

In early 2025, Electron’s npm ecosystem repos (under the `@electron/` and `@electron-forge/` namespaces) will move to Node.js 22 as the minimum supported version.

---

> 
> In early 2025, Electron’s npm ecosystem repos (under the `@electron/` and `@electron-forge/` namespaces) will move to Node.js 22 as the minimum supported version.

Reviewer 2: ```suggestion
```

This is duplicated in the rendered version, so let's just remove it.

### What does this mean?

In the past, packages in Electron’s npm ecosystem (Forge, Packager, etc) have supported Node versions for as long as possible, even after a version has reached its End-Of-Life (EOL) date. This is done to make sure we don’t fragment the ecosystem—we understand that many projects depend on older versions of Node, and we don’t want to risk stranding those projects unless there was a pressing reason to upgrade.

Over time, maintenance of this policy has become increasingly difficult for a few reasons:

- Lack of official Node.js 14 macOS ARM64 builds requires us to maintain CI infrastructure workarounds to provide full test coverage.
- `engines` requirements for upstream package dependencies have moved forward, making it increasingly difficult to resolve supply chain security issues with dependency bumps.

> Additionally, newer versions of Node.js have included many improvements that we would like to leverage, such as runtime-native common utilities (e.g. `fs.glob` and `util.parseArgs`) and entire new batteries-included modules (e.g. `node:test`, `node:sqlite`).

Reviewer 2: ```suggestion
Additionally, newer versions of Node.js have included many improvements that we would like to leverage, such as runtime-native common utilities (e.g. [`fs.glob`](https://nodejs.org/api/fs.html#fsglobpattern-options-callback) and [`util.parseArgs`](https://nodejs.org/api/util.html#utilparseargsconfig)) and entire new batteries-included modules (e.g. [`node:test`](https://nodejs.org/api/test.html), [`node:sqlite`](https://nodejs.org/api/sqlite.html)).
```

Reading over this, I think adding links would be a good enhancement
